src/policy/compiler.rs

```diff
@@ -1613,11 +1598,13 @@ mod tests {
         );
         // For legacy too..
         let (keys, _) = pubkeys_and_a_sig(68);
-        let keys: Vec<Arc<Concrete<bitcoin::PublicKey>>> = keys
-            .iter()
-            .map(|pubkey| Arc::new(Concrete::Key(*pubkey)))
-            .collect();
-        let thresh_res = Concrete::Thresh(keys.len() - 1, keys).compile::<Legacy>();
+        let thresh = Threshold::from_iter(
+            keys.len() - 1,
+            keys.iter().map(|pubkey| Arc::new(Concrete::Key(*pubkey))),
+        )
+        .unwrap();
```

Comment:
I notice there's a few unwraps() in this PR.  Just curious if there's a reason to do this instead of returning an Error type.

Reply:
These are tests. `unwraps` are common in test and that is how we test most things.

Reply:
Oops, I had meant to place this comment on the  [`generate_combination()`](https://github.com/rust-bitcoin/rust-miniscript/pull/674/files/1345fb2de2caa7d2def4acbf26757c872d62c8d6#diff-c88a4b1ba51d4a4eb96d972ee202e5b484fc29f573aee6655bd3982bf4a37ee6R1041) changes.  Heh yeah, I get using unwrap on the tests though.

Reply:
Yeah, that was one occurrence which was a safe unwrap because of miniscript invariants around `Threshold`. We could add a expect with a message with the reasoning, but there was a short comment the commit message explaining that.

Reply:
I added an `expect`. There was also a `debug_assert` which would have triggered before the `unwrap` did, but better to be explicit.